newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/FormatExpressionCall.js

```diff
@@ -14,7 +14,8 @@ const filterOutCodeOnlyParameters = (
 
     return (
       index < parametersCount &&
-      !expressionMetadata.getParameter(index).isCodeOnly()
+      !expressionMetadata.getParameter(index).isCodeOnly() &&
+      parameter !== ''
```

Comment:
This is unfortunately a bit too naive.
If you insert a new expression like MouseX, you'll be asked for the layer name and camera number (though it's not used currently in GDevelop because all layers have a single camera).

If I enter nothing for the layer, but a number for the camera (0), this will result in the current situation in:
`MouseX(,0)` - which is not valid, but this is ok because what you entered is not valid (if you enter a camera number, you must enter a layer name).

With your changes, this will now make:
`MouseX(0)` which looks valid but is not, because you're passing a number instead of a layer name.

(Actually, note that GDevelop is forgiving and can understand MouseX(,0)... but MouseX(0) is definitely wrong)

Re-read my comment for the proper logic: https://github.com/4ian/GDevelop/issues/1533#issuecomment-599204949

Reply:
So we have to pass the parameters that comes after required parameters.
What I understand is parameter.isOptional() && parameter.isEmpty() is the condition to not add a parameter to the list. I'll use the app more and learn the behaviour